src/include/stir/recon_buildblock/niftypet_projector/NiftyPETHelper.h

```diff
@@ -0,0 +1,161 @@
+
+    /// Convert NiftyPET image to STIR image
+    static void convert_image_niftyPET_to_stir(DiscretisedDensity<3,float> &stir, const std::vector<float> &np_vec);
+
+    /// Convert STIR proj data to NiftyPET proj data
+    void convert_proj_data_stir_to_niftyPET(std::vector<float> &np_vec, const ProjData& stir) const;
+
+    /// Convert STIR viewgram to NiftyPET
+    void convert_viewgram_stir_to_niftyPET(std::vector<float> &np_vec, const Viewgram<float>& viewgram) const;
+
+    /// Convert NiftyPET proj data to STIR proj data
+    void convert_proj_data_niftyPET_to_stir(ProjData &stir_sptr, const std::vector<float> &np_vec) const;
+
+    /// Remove gaps from sinogram. Do some unavoidable const_casting as the wrapped methods don't use const
+    void remove_gaps(std::vector<float> &sino_no_gaps, const std::vector<float> &sino_w_gaps) const;
+
+    /// Put gaps into sinogram. Do some unavoidable const_casting as the wrapped methods don't use const
+    void put_gaps(std::vector<float> &sino_w_gaps, const std::vector<float> &sino_no_gaps) const;
+
+    /// Back project. Do some unavoidable const_casting as the wrapped methods don't use const
+    void back_project(std::vector<float> &image, const std::vector<float> &sino_no_gaps) const;
+
+    /// Forward project, returns sinogram without gaps. Do some unavoidable const_casting as the wrapped methods don't use const
+    void forward_project(std::vector<float> &sino_no_gaps, const std::vector<float> &image) const;
+
+    /// Create a STIR sinogram
+    static shared_ptr<ProjData> create_stir_sino();
+
+    /// Listmode to sinogram
+    void lm_to_proj_data(shared_ptr<ProjData> &prompts_sptr, shared_ptr<ProjData> &delayeds_sptr,
+                         shared_ptr<ProjData> &randoms_sptr, shared_ptr<ProjData> &norm_sptr,
+                         const int tstart, const int tstop,
+                         const std::string &lm_binary_file, const std::string &norm_binary_file="") const;
+
+private:
+
+    /// Check that set up has been run before returning data
+    void check_set_up() const;
+
+    /// Permute the data
+    void permute(std::vector<float> &output_array, const std::vector<float> &orig_array, const unsigned output_dims[3], const unsigned *permute_order) const;
+
+    /// Convert 3d niftypet proj data index to 1d
+    unsigned convert_niftypet_proj_3d_to_1d_idx(const unsigned ang, const unsigned bins, const unsigned sino) const;
+
+    /// Convert 1d niftypet proj data index to 3d
+    void convert_niftypet_proj_1d_to_3d_idx(unsigned &ang, unsigned &bins, unsigned &sino, const unsigned idx) const;
+
+    float _niftypet_to_stir_ratio;
+    bool _already_set_up;
+    char _span;
+    char _devid;
+    shared_ptr<Cnst> _cnt_sptr;
+    int _nsinos;
+    char _att;
+    std::vector<int> _isub;
+    bool _verbose;
+    Scanner::Type _scanner_type;
+    shared_ptr<txLUTs> _txlut_sptr;
+    shared_ptr<axialLUT> _axlut_sptr;
+
+    float *_crs;
```

Comment:
please add a comment what's going to happen with these bare points. they're scary

Reply:
replaced with vectors